plutus-core/testlib/PlutusCore/Generators/Hedgehog/Builtin.hs

```diff
@@ -145,3 +157,17 @@ genConstr depth =
         <*> Gen.list
             (Range.linear 0 5)
             (genData (depth - 1))
+
+genBls12_381_G1_Element :: Gen Crypto.BLS12_381.G1.Element
+genBls12_381_G1_Element = Crypto.BLS12_381.G1.hashToGroup <$> genByteString
+
+genBls12_381_G2_Element :: Gen Crypto.BLS12_381.G2.Element
+genBls12_381_G2_Element = Crypto.BLS12_381.G2.hashToGroup <$> genByteString
+
+genBls12_381_MlResult :: Gen Crypto.BLS12_381.Pairing.MlResult
+genBls12_381_MlResult = do
+  p1 <- genBls12_381_G1_Element
+  p2 <- genBls12_381_G2_Element
+  case Crypto.BLS12_381.Pairing.millerLoop p1 p2 of
+    Left e  -> error $ "BLS12-381 Miller loop failed in genBls12_381_MlResult" ++ (show e)
```

Comment:
`millerLoop` should succeed on all points in G1 and G2, but I think the C library function may fail if you give it points which are on the curve but not in the subgroup (which we should never be able to get anyway).  I suppose the Haskell function that calls the C function can't just fail if the points aren't in the group, so it has to return an `Either` and then we have to deal with it here.  I thought it would probably be OK to have `error` in tests and benchmarks.

Reply:
Ahá, very good catch. This is indeed a leftover. Before we did have the possibility of failure, because we allowed G1Elements to not be part of the subgroup, so we had to check that. Now that we enforce at type level that G1Elements are part of the subgroup, the millerLoop will not fail. I'll change that in base.

Reply:
The merged version of the bindings does not return an `Either` here anymore.

Reply:
I have now fixed that.